src/lib/scheduler/processing_unit.cpp

```diff
@@ -6,6 +6,8 @@
 #include "uid_allocator.hpp"
 #include "worker.hpp"
 
+#define MAX_WORKERS_PER_CORE 200
```

Comment:
I picked the number arbitrarily (it worked quite well in my experiments). Not sure how to systematically pick the best number. I easily surpassed it. When you create a lot of table scan operations each task spawns subtasks and waits for them. Now, the original thread is blocked due to waiting. The scheduler now spawns a new thread per subtask (because the original is blocked). If you schedule enough table scans, you'll easily surpass 200 and eventually the whole process becomes overloaded.

Not really sure how to solve this with the current task model other than to limit the number of workers. An async model with a single thread per core and task yielding might solve this.

Reply:
This is a completely unrelated tuning issue. Interesting to see that you had more than 200 workers running. Maybe just add a comment that this number is arbitrarily chosen but has been reached in benchmarks.